babel/src/main/java/org/apache/calcite/sql/babel/SqlHexCharStringLiteral.java

```diff
@@ -0,0 +1,136 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.calcite.sql.babel;
+
+import org.apache.calcite.sql.SqlColumnAttributeCharacterSet.CharacterSet;
+import org.apache.calcite.sql.SqlLiteral;
+import org.apache.calcite.sql.SqlUtil;
+import org.apache.calcite.sql.SqlWriter;
+import org.apache.calcite.sql.parser.SqlParserPos;
+import org.apache.calcite.sql.parser.SqlParserUtil;
+import org.apache.calcite.sql.type.SqlTypeName;
+import org.apache.calcite.util.NlsString;
+
+import java.util.Locale;
+
+import static org.apache.calcite.util.Static.RESOURCE;
+
+/**
+ * Parse tree for {@code SqlHexCharStringLiteral} expression.
+ */
+public class SqlHexCharStringLiteral extends SqlLiteral {
+
+  final HexCharLiteralFormat format;
+  final CharacterSet charSet;
+
+  /**
+   * Creates a {@code SqlHexCharStringLiteral}.
+   * @param hex           String that contains the hex characters
+   * @param pos           Parser position, must not be null
+   * @param charSetString Character set string, can be null
+   * @param formatString  Format of the hex char literal
+   */
+  public SqlHexCharStringLiteral(final String hex, final SqlParserPos pos,
+      final String charSetString, final String formatString) {
+    super(new NlsString(hex, null, null), SqlTypeName.CHAR, pos);
+    if (charSetString == null) {
+      this.charSet = null;
+    } else {
+      String normalizedCharSetString = SqlParserUtil
+          .trim(charSetString, " ")
+          .toUpperCase(Locale.ROOT);
+      switch (normalizedCharSetString) {
+      case "LATIN":
+        charSet = CharacterSet.LATIN;
+        break;
+      case "UNICODE":
+        charSet = CharacterSet.UNICODE;
+        break;
+      case "GRAPHIC":
+        charSet = CharacterSet.GRAPHIC;
+        break;
+      case "KANJISJIS":
+        charSet = CharacterSet.KANJISJIS;
+        break;
+      default:
+        throw SqlUtil.newContextException(pos,
+            RESOURCE.unknownCharacterSet(charSetString));
+      }
+    }
+    switch (formatString) {
+    case "XC":
+      format = HexCharLiteralFormat.XC;
+      break;
+    case "XCV":
+      format = HexCharLiteralFormat.XCV;
+      break;
+    default:
+      format = HexCharLiteralFormat.XCF;
+      break;
+    }
+  }
+
+  @Override public void unparse(final SqlWriter writer, final int leftPrec,
+      final int rightPrec) {
+    if (charSet != null) {
+      writer.print("_");
+      switch (this.charSet) {
+      case LATIN:
+        writer.keyword("LATIN");
+        break;
+      case UNICODE:
+        writer.keyword("UNICODE");
+        break;
+      case GRAPHIC:
+        writer.keyword("GRAPHIC");
+        break;
+      case KANJISJIS:
+        writer.keyword("KANJISJIS");
+        break;
+      }
+    }
+    writer.keyword(value.toString());
```

Comment:
nit: I think a better method to use here might be writer.literal(). This is what is used elsewhere to unparse values that are not actually keywords:

https://github.com/googleinterns/calcite/blob/5d373516fe48ad3a8c97a7c69a90bca20a4e480a/core/src/main/java/org/apache/calcite/sql/SqlNumericLiteral.java#L77

https://github.com/googleinterns/calcite/blob/5d373516fe48ad3a8c97a7c69a90bca20a4e480a/core/src/main/java/org/apache/calcite/sql/SqlLiteral.java#L722

Reply:
Fix in https://github.com/googleinterns/calcite/pull/155/commits/6a66d30df6c0e6d84e76befa55137ee96d66de4f